src/hypergraph/hypergraph.cpp

```diff
@@ -570,35 +613,39 @@ void remove_free_nets(pmondriaan::hypergraph& H, size_t max_size) {
  * Simplifies all duplicate nets.
  */
 void simplify_duplicate_nets(pmondriaan::hypergraph& H) {
-    std::map<std::vector<long>, int> edge_counts;
-    std::map<std::vector<long>, long> ids;
-
-    for (auto n : H.nets()) {
-        edge_counts[n.vertices()] += n.cost();
-        ids[n.vertices()] = n.id();
-    }
-
-    std::unordered_set<long> remove_nets;
-
-    for (auto n = 0u; n < H.nets().size(); n++) {
-        if(ids[H.nets()[n].vertices()] != H.nets()[n].id()) {
-            remove_nets.insert(H.nets()[n].id());
-        }
-        else {
-            H.nets()[n].set_cost(edge_counts[H.nets()[n].vertices()]);
+    // Initialize maps
+    boost::unordered_map<std::vector<long>, long, VectorHasher> edge_counts;
+    std::vector<long> remove_nets;
+
+    // Simplify nets that are not too large.
+    // Note that smaller hyperedges are more common in most cases, if not then simplification
+    // is of questionable use anyways.
+    // Other than this we just sum the costs for some set of vertices found to map to
+    // a set of identifyer long integers in the pins of a hyperedges.
+    for (auto& n : H.nets()) {
+        if(n.vertices().size() < 100) {
+            if(edge_counts.find(n.vertices()) == edge_counts.end()) {
+                edge_counts[n.vertices()] = n.id();
+            }
+            else {
+                auto& net = H.nets()[H.local_id_net(edge_counts[n.vertices()])];
+                net.set_cost(n.cost() + net.cost());
+                remove_nets.push_back(n.id());
+            }
         }
     }
 
     for (auto n : remove_nets) {
         H.remove_net_by_index(H.local_id_net(n));
     }
+
 }
 
 /**
- * Break triples.
+ * Break triples, this does not account for the policy around the usage..
  */
 void break_triples(pmondriaan::hypergraph& H) {
-    std::unordered_set<long> remove_nets;
+    std::vector<long> remove_nets;

```

Comment:
I thought underordered_set should be more efficient, is that not the case?

Reply:
I believe that in some cases the overhead of unordered sets were quit big, while the one check I added to net removal to see if the net was found seemed to be more efficient experimentally.